File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        freq = {}
        for i in range(len(nums)):
            required = target-nums[i]
            if required in freq:
                return [freq[required],i]
            freq[nums[i]] = i
        return [-1,-1]

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        const map = new Map()
        for(let i=0;i<numbers.length;i++){
            let req = target - numbers[i]
            if(map.has(req)){
                return [map.get(req)+1,i+1]
            }
            map.set(numbers[i],i)            
        }
    }
}
